### PR TITLE
IoT Edge deployment validates content with respect to schema

### PR DESCRIPTION
**Pull request by digimaun**
Addresses issue #84 

* Both `az iot edge deployment create` and `az iot edge set-modules` will *always* validate v2 input content with respect to the [IoT Edge deployment schema v2](https://schemastore.azurewebsites.net/schemas/json/azure-iot-edge-deployment-2.0.json).
* Instead of pulling from an external resource, the schema is stored in IoT extension package assets. The stored schema asset is used for validation.
* Any validation issues will be bubbled up.

image

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [X] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [X] Have **all** unit **and** integration tests passed locally? i.e. `pytest -s <project root>/azext_iot/tests`
- [X] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.

### CODE REVIEW

File: azext_iot/tests/test_config_modules_content.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
             },
             "$edgeHub": {
                 "properties.desired": {
-                    "schema": "1.0",
+                    "schemaVersion": "1.0",
```
Comment (chieftn) [inline azext_iot/tests/test_config_modules_content.json]: nice catch.

File: azext_iot/operations/hub.py [hunk 2]
```diff
@@ -617,6 +622,28 @@ def _process_config_content(content, content_type='module'):
     return content
 
 
+def _validate_payload_schema(content):
+    from jsonschema import validate
+    from jsonschema.exceptions import ValidationError, SchemaError
+    from azext_iot._constants import EDGE_DEPLOYMENT_SCHEMA_2_PATH as schema_path
```
Comment (chieftn) [inline azext_iot/operations/hub.py]: schema file version 2.0 has image pull policy to the Edge Agent schema as an optional parameters.  Is there any impact to it's use in the cli?

File: azext_iot/tests/test_iot_ext_unit.py [hunk 3]
```diff
@@ -820,13 +827,23 @@ def test_config_create_invalid_args(self, serviceclient, req, arg):
                                                        priority=req['priority'], labels=req['labels'])
 
     @pytest.mark.parametrize("req", [
-        (generate_device_config(content_type='device'))
+        (generate_device_config(content_type='device')),
```
Comment (YingXue) [inline azext_iot/tests/test_iot_ext_unit.py]: seems like a trailing comma

File: setup.py [hunk 4]
```diff
@@ -40,28 +42,27 @@
...
+    author="Microsoft",
+    author_email="iotupx@microsoft.com",  # +@digimaun
+    url="https://github.com/azure/azure-iot-cli-extension",
     classifiers=CLASSIFIERS,
-    packages=find_packages(),
-    package_data={'azext_iot': ['azext_metadata.json', 'digicert.pem']},
-    install_requires=DEPENDENCIES
+    packages=find_packages(exclude=["tests", "tests.*", "scripts"]),
+    package_data={"azext_iot": ["azext_metadata.json", "digicert.pem", "assets/edge-deploy-2.0.schema.json"]},
+    install_requires=DEPENDENCIES,
```
Comment (YingXue) [inline setup.py]: another trailing comma, unless it is intentional

File: azext_iot/operations/hub.py [hunk 5]
```diff
@@ -617,6 +622,28 @@ def _process_config_content(content, content_type='module'):
...
+
+    try:
+        logger.info("Validating deployment payload...")
```
Comment (YingXue) [inline azext_iot/operations/hub.py]: NIT: I may move this logger after the following if block. Seems to make more sense to me to see this kind of in progress log only if it actually validating.

File: azext_iot/tests/test_iot_ext_int.py [hunk 6]
```diff
@@ -14,37 +14,44 @@
...
+from azext_iot.common.utility import (
+    validate_min_python_version,
+    execute_onthread,
+    calculate_millisec_since_unix_epoch_utc,
+)
 from azext_iot._constants import DEVICE_DEVICESCOPE_PREFIX
 
 # Add test tools to path
-sys.path.append(os.path.abspath(os.path.join('.', 'iotext_test_tools')))
+sys.path.append(os.path.abspath(os.path.join(".", "iotext_test_tools")))
```
Comment (YingXue) [inline azext_iot/tests/test_iot_ext_int.py]: are all the single quote to double quote a linter change?

File: azext_iot/tests/test_iot_ext_unit.py (hunk 3 above)
Comment (digimaun) [reply azext_iot/tests/test_iot_ext_unit.py]: Yea the trailing commas are coming from auto-formatting using Python Black...it's an amazing tool.

File: azext_iot/tests/test_iot_ext_int.py (hunk 6 above)
Comment (digimaun) [reply azext_iot/tests/test_iot_ext_int.py]: Yea it's coming from python black autoformatter.

File: setup.py (hunk 4 above)
Comment (digimaun) [reply setup.py]: Yea the trailing commas are coming from auto-formatting using Python Black...it's an amazing tool.

File: azext_iot/operations/hub.py (hunk 2 above)
Comment (digimaun) [reply azext_iot/operations/hub.py]: Any optional parameters like the imagePullPolicy would assume the service chooses a sensible default. So only missing required elements would prevent the CLI from submitting the deployment payload.

File: azext_iot/operations/hub.py (hunk 5 above)
Comment (digimaun) [reply azext_iot/operations/hub.py]: Moved.

**Pull request merged**